cmd/csalt/main.go

```diff
@@ -248,6 +281,43 @@ func apiFromArgs(args Args) (*userapi.CacophonyUserAPI, string, error) {
 	return api, saltPrefix, nil
 }
 
+func checkForDuplicates(devices *userapi.DeviceResponse) error {
+	nameMap := make(map[string][]userapi.Device)
+	duplicateNames := make([]string, 0, 1)
+	for _, device := range devices.NameMatches {
+		if _, ok := nameMap[device.DeviceName]; !ok {
+			nameMap[device.DeviceName] = []userapi.Device{device}
+		} else {
+			nameMap[device.DeviceName] = append(nameMap[device.DeviceName], device)
+			duplicateNames = append(duplicateNames, device.DeviceName)
+		}
+	}
+	if len(duplicateNames) > 0 {
+		fmt.Printf("DeviceName Query found %v Duplicate Device please specify group:devicename\n", len(duplicateNames))
```

Comment:
This could be tightened up: `Found %v ambigous devices. Please specify these devices in full group:devicename form.\n`